### ISSUE TITLE
zIndex not working

### ISSUE THREAD

**Issue by buog** [react-native-mapbox-gl/maps#1627]
**Describe the bug**
I'm trying to build a LIVE map and I noticed that the zIndex of the react-native mapbox gl doesn't work when the render is in a different state. When you render something after, it stays underneath. Is there anyway for certain things to always be on top? For example, callout should always be on top no matter what.

**To Reproduce**

```
<MapboxGL.MapView
        style={{flex: 1}}
        styleURL={'mapbox://styles/gigajungkyu/ckt962t3a19zh17pfq1sr7z3f'}
        pitchEnabled={false}
        rotateEnabled={false}>
        <MapboxGL.PointAnnotation
          id={'1'}
          key={'1'}
          style={{zIndex: 1}}
          coordinate={[-122.849, 49.1]}>
          <TouchableOpacity
            style={{zIndex: 1}}
            onPress={() => setDisplay((prevState) => !prevState)}>
            <View
              style={{width: 100, height: 100, backgroundColor: 'red'}}></View>
          </TouchableOpacity>
          <MapboxGL.Callout title={'You'} />
        </MapboxGL.PointAnnotation>
        {display && (
          <MapboxGL.PointAnnotation
            id={'2'}
            key={'2'}
            style={{zIndex: 99999999}}
            coordinate={[-103.349, 49.1]}>
            <TouchableOpacity style={{zIndex: 5}}>
              <View
                style={{
                  width: 100,
                  height: 100,
                  backgroundColor: 'blue',
                }}></View>
            </TouchableOpacity>
            <MapboxGL.Callout title={'You'} />
          </MapboxGL.PointAnnotation>
        )}
      </MapboxGL.MapView>
```

Expected behavior
Blue box be infront of red box

**Versions (please complete the following information):**
Platform: iOS
Platform OS: iOS 11
Emulator/ Simulator: yes
Dev OS: macOS 12.0.1 (21A559)
react-native-mapbox-gl Version: off master branch (8.4.0)
Mapbox GL version: 5.9.0
React Native Version: 0.66.0

**Comment by ferdicus** [comment]
Dear @buog, thanks for opening the issue.  
Please do me a favor and do not simply close them without adding a clarifying comment. 

Reasoning:
* other people may find your ticket through a search
* now they see it is closed and they are wondering "what did they do to fix this issue?!"

So, even if you figured out what the problem was/ noticed this wasn't an issue after all

just drop a line when closing

Thanks in advance 🙇🏿

**Comment by RyanTG** [comment]
This is old and mysteriously closed, but even today with version `10.0.12`  I cannot get `style={{zIndex: whatever}}` to affect the order of the PointAnnotations (on either iOS or Android). 

@buog Any insight? Did you resolve this?
